plutus-core/plutus-core/src/PlutusCore/Parser/Builtin.hs

```diff
@@ -6,80 +6,34 @@ module PlutusCore.Parser.Builtin where
 
 import Data.Text qualified as T
 import Data.Text.Internal.Read (hexDigitToInt)
-import PlutusPrelude
-import Text.Megaparsec hiding (ParseError, State, parse, some)
+import PlutusPrelude (Word8)
+import Text.Megaparsec (MonadParsec (takeWhileP), choice, many, manyTill)
 import Text.Megaparsec.Char (char, hexDigitChar)
 import Text.Megaparsec.Char.Lexer qualified as Lex hiding (hexadecimal)
 
 import Data.ByteString (pack)
+import Data.Map (Map, empty, insert, lookup)
 import PlutusCore.Default
-import PlutusCore.Parser.ParserCommon (Parser, lexeme, symbol, whitespace)
+import PlutusCore.Parser.ParserCommon (Parser, isIdentifierChar, lexeme, symbol, whitespace)
 import PlutusCore.Parser.Type (defaultUniType)
+import PlutusCore.Pretty (display)
+import Prelude hiding (lookup)
 
+mkCachedBuiltin :: [DefaultFun] -> Map T.Text DefaultFun -> Map T.Text DefaultFun
+mkCachedBuiltin (hdFns:tlFns) builtinMap =
+    mkCachedBuiltin tlFns (insert (display hdFns) hdFns builtinMap)
+mkCachedBuiltin [] builtinMap = builtinMap
```

Comment:
Nitpicking (i.e. not important at all): instead of writing the recursion manually you define `mkCachedBuiltin` as `Map.fromList . map (\fun -> (display fun, fun))` or something like that.

Reply:
yeah HLint told me to do that too but I thought it's harder to read. Maybe not. I can change that. I was gonna open a follow up PR for the error anyway.